internal/node/telemetry.go

```diff
@@ -160,11 +167,10 @@ func (t *Telemetry) initMeterProvider() error {
 			),
 		)
 
-		defer func() {
-			if err := meterProvider.(*metricsdk.MeterProvider).Shutdown(context.Background()); err != nil {
-				t.log.Error("failed to shutdown OTel meter provider", slog.Any("err", err))
-			}
-		}()
+		if _, ok := meterProvider.(*metricsdk.MeterProvider); ok {
+			t.providerShutdown = meterProvider.(*metricsdk.MeterProvider).Shutdown
+		}
+
 	} else {
 		meterProvider = noop.NewMeterProvider()
```

Comment:
Should we keep a reference around to `meterProvider` instead of `providerShutdown`? Then in `Shutdown()` we can just call `t.meterProvider.Shutdown(ctx)`?